Source/Factors/OktaFactorPush.swift

```diff
@@ -0,0 +1,210 @@
+
+    public var activation: EmbeddedResponse.Factor.Embedded.Activation? {
+        get {
+            return factor.embedded?.activation
+        }
+    }
+
+    public var activationLinks: LinksResponse? {
+        get {
+            return factor.embedded?.activation?.links
+        }
+    }
+
+    public var qrCodeLink: LinksResponse.QRCode? {
+        get {
+            return factor.embedded?.activation?.links?.qrcode
+        }
+    }
+
+    public var sendLinks: [LinksResponse.Link]? {
+        get {
+            return factor.embedded?.activation?.links?.send
+        }
+    }
+
+    public func canSendPushCodeViaSms() -> Bool {
+        guard let sendLinkArray = factor.links?.send else {
+            return false
+        }
+        
+        for link in sendLinkArray {
+            if link.name == "sms" {
+                return true
+            }
+        }
+        
+        return false
+    }
+
+    public func codeViaSmsLink() -> LinksResponse.Link? {
+        guard let sendLinkArray = factor.links?.send else {
+            return nil
+        }
+        
+        for link in sendLinkArray {
+            if link.name == "sms" {
+                return link
+            }
+        }
+        
+        return nil
+    }
+    
+    public func canSendPushCodeViaEmail() -> Bool {
+        guard let sendLinkArray = factor.links?.send else {
+            return false
+        }
+        
+        for link in sendLinkArray {
+            if link.name == "email" {
+                return true
+            }
+        }
+        
+        return false
+    }
+
+    public func codeViaEmailLink() -> LinksResponse.Link? {
+        guard let sendLinkArray = factor.links?.send else {
+            return nil
+        }
+        
+        for link in sendLinkArray {
+            if link.name == "email" {
+                return link
+            }
+        }
+        
+        return nil
+    }
+
+    public func sendActivationLinkViaSms(with phoneNumber:String,
+                                         onSuccess: @escaping () -> Void,
+                                         onError: @escaping (_ error: OktaError) -> Void) {
+        guard canSendPushCodeViaSms() else {
+            onError(OktaError.wrongStatus("Can't find 'send' link in response"))
+            return
+        }
+
+        guard responseDelegate != nil else {
+            onError(OktaError.invalidParameters("Empty responseDelegate"))
+            return
+        }
+
+        restApi?.sendActivationLink(link: codeViaSmsLink()!,
+                                    stateToken: stateToken,
+                                    phoneNumber: phoneNumber,
+                                    completion: { result in
+                                        switch result {
+
+                                        case .error(let error):
+                                            onError(error)
+                                            return
```

Comment:
Minor: `return` statement is not required here since Swift switch does not fall through the bottom of each case and into the next one by default.